lib/utils.js

```diff
@@ -251,9 +251,43 @@ function getPackageVersion (pkgName) {
   return pkgInfo.version;
 }
 
+/**
+ * Pulls the initial values of Appium settings from the given capabilities argument.
+ * Each setting item must satisfy the following format:
+ * `setting[setting_name]: setting_value`
+ * The capabilities argument itself gets mutated, so it does not contain parsed
+ * settings anymore to avoid further parsing issues.
+ * Check
+ * https://github.com/appium/appium/blob/master/docs/en/advanced-concepts/settings.md
+ * for more details on the available settings.
+ *
+ * @param {?Object} caps - Capabilities dictionary. It is mutated if
+ * one or more settings have been pulled from it
+ * @returns {Object} - An empty dictionary if the given caps contains no
+ * setting items or a dictionary containing parsed Appium setting names along with
+ * their values.
+ */
+function pullSettings (caps) {
+  if (!_.isPlainObject(caps) || _.isEmpty(caps)) {
+    return {};
+  }
+
+  const result = {};
+  for (const [key, value] of _.toPairs(caps)) {
+    const match = /\bsettings\[(\S+)\]$/.exec(key);
```

Comment:
Maybe `const [, settingName] =  /\bsettings\[(\S+)\]$/.exec(key);`

Reply:
this will throw if match is not an array

Reply:
Can we still have a variable called `settingName` to make it more clear what's being parsed.  

`const [, settingName] = /\bsettings\[(\S+)\]$/.exec(key) || []`;